VRDR.Tests/Filtering_Should.cs

```diff
@@ -0,0 +1,341 @@
+            BaseMessage postFilteredBaseMessage;
+            
+            try
+            {
+                postFilteredBaseMessage = NCHSFilterService.filterMessage(preFilteredBaseMessage);
+                BaseMessage.Parse(postFilteredBaseMessage.ToJson());
+            }
+            catch (Exception e)
+            {
+                throw new Exception(e.Message);
+            }
+            
+            DeathRecord preFilteredDeathRecord = getDeathRecordFromMessage(preFilteredBaseMessage);
+            DeathRecord postFilteredDeathRecord = getDeathRecordFromMessage(postFilteredBaseMessage);
+            
+            Assert.Equal(preFilteredDeathRecord.ResidenceWithinCityLimits, postFilteredDeathRecord.ResidenceWithinCityLimits);
+        }
+        
+        [Fact]
+        public void PreFilteredFileEqualsFilteredFile()
+        {
+            List<string> passedFields = new List<string>();
+            List<string> failedFields = new List<string>();
+            
+            BaseMessage preFilteredBaseMessage = BaseMessage.Parse(File.ReadAllText("./fixtures/filter/Pre-filtered-file.json"));
+            BaseMessage postFilteredBaseMessage;
+
+            try
+            {
+                postFilteredBaseMessage = allFieldsFilterService.filterMessage(preFilteredBaseMessage);
+                BaseMessage.Parse(postFilteredBaseMessage.ToJson());
+            }
+            catch (Exception e)
+            {
+                throw new Exception(e.Message);
+            }
+
+            DeathRecord preFilteredDeathRecord = getDeathRecordFromMessage(preFilteredBaseMessage);
+            DeathRecord postFilteredDeathRecord = getDeathRecordFromMessage(postFilteredBaseMessage);
+            
+            foreach (var property in typeof(DeathRecord).GetProperties())
+            {
+                // Console.WriteLine("==================================");
+                object preFilteredObject = JsonConvert.SerializeObject(property.GetValue(preFilteredDeathRecord), Formatting.Indented);
+                object postFilteredObject = JsonConvert.SerializeObject(property.GetValue(postFilteredDeathRecord), Formatting.Indented);
+
+                if (!preFilteredObject.Equals(postFilteredObject))
+                {
+                    failedFields.Add(property.Name);
+                    
+                    // Console.WriteLine($"{property.Name}");
+                    // Console.WriteLine(preFilteredObject);
+                    // Console.WriteLine(postFilteredObject);
+                }
+                else
+                {
+                    passedFields.Add(property.Name);
+                    
+                    // Console.WriteLine($"{property.Name}");
+                    // Console.WriteLine(preFilteredObject);
+                    // Console.WriteLine(postFilteredObject);
+                }
+            }
+            
+            Console.WriteLine("=============Field filtering test=============");
```

Comment:
There are no assertions here, this test needs to have assertions so that it fails if the filtered passed and failed fields are not what is expected.  The same for the subsequent tests which are lacking assertions.

Reply:
For some of these tests, we aren't testing a single assertion. Some test that a valid BaseMessage is generated from the filter.cs class. I see your point with the ones that have success and failure arrays, I will add them there.
What would you suggest for the test that test multiple assertions?

Reply:
You can add all the assertions in the single test, so that if any assertions fail, the whole test fails.  For instance in Messaging_Should.cs you can see most of the tests have many assertions per test.  For instance you could do individual field assertions for important fields, and total field count checks for everything.

For the BaseMessage.Parse runs, you can do:
1. Remove the try/catch blocks as they shouldn't be necessary around the test, for instance see Messaging_Should::CreateUpdateFromJson()
2. Add an assertion about the returned message, such as that it isn't null, has the correct certificate number, just something to tell you the message was parsed properly after the filter service processed it

For at least one test, make sure the filtering fails on an invalid value, such as is done with Messaging_Schold.SelectMessageType() where it does Assert.Throws() calls to make sure invalid messages generate exceptions on parsing, for filtering you want something similar.

Because we have a number of upstream consumers of the library, we do more than the average amount of tests to give higher confidence in the released software.